lib/pages/events_page.dart

```diff
@@ -14,6 +16,7 @@ class EventsPage extends StatefulWidget {
 
 class _EventsPageState extends State<EventsPage> {
   final _refreshIndicatorKey = new GlobalKey<RefreshIndicatorState>();
+  static const DATE_FORMAT = 'EEE, MMM d';
```

Comment:
Date formatting stuff is fine to be at the provider imo
i meant that we could leave out the index == 0 / index == 1 logic to evens_page only
so 
`.contains(DateFormat(DATE_FORMAT).format(DateTime.now())))` stuff was great as you did before in the provider
no need to move it here in the page widget imo

sorry i wasn't that clear

Reply:
Can you check again :D I'm not sure if I understand correctly.

Reply:
It's exactly what i meant! :) 👍